Weather App/WeathersList/viewModel/CitiesRepository.swift

```diff
@@ -0,0 +1,48 @@
+import CoreLocation
+import Foundation
+import Keys
+
+class CitiesRepository {
+    public let session: URLSession
+    let requestUrl: String
+
+    init() {
+        session = URLSession(configuration: .default)
+        guard let basicURL = Bundle.main.object(forInfoDictionaryKey: "WeatherRequestURL") as? String else {
+            fatalError("Incorrect Open Weather URL ")
+        }
+
+        requestUrl = basicURL
+    }
+
+    public func getWeatherForCity(_ city: City) async -> CityWeather? {
+        guard let url = URL(string: requestUrl + "?q=\(city.plainName)" +
+            "&appid=\(Weather_AppKeys().openWeatherApiKey)&units=metric")
+        else {
+            return nil
+        }
+        return await getWeather(url: url, isFromLocation: false)
+    }
+
+    public func getWeatherForCoordinate(_ coordinate: CLLocationCoordinate2D) async -> CityWeather? {
```

Comment:
do you need public access here? I think, internal will be enough

Reply:
👍